packages/purpose-process/test/api/activatePurposeVersion.test.ts

```diff
@@ -0,0 +1,143 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { describe, it, expect, vi, beforeEach } from "vitest";
+import { generateId } from "pagopa-interop-models";
+import {
+  generateToken,
+  getMockPurpose,
+  getMockPurposeVersion,
+} from "pagopa-interop-commons-test";
+import { authRole } from "pagopa-interop-commons";
+import request from "supertest";
+import { purposeApi } from "pagopa-interop-api-clients";
+import { api, purposeService } from "../vitest.api.setup.js";
+import { purposeVersionToApiPurposeVersion } from "../../src/model/domain/apiConverter.js";
+import {
+  agreementNotFound,
+  missingRiskAnalysis,
+  organizationIsNotTheConsumer,
+  organizationIsNotTheProducer,
+  organizationNotAllowed,
+  purposeNotFound,
+  purposeVersionNotFound,
+  riskAnalysisValidationFailed,
+} from "../../src/model/domain/errors.js";
+
+describe("API POST /purposes/{purposeId}/versions/{versionId}/activate test", () => {
+  const mockPurposeVersion = getMockPurposeVersion();
+  const mockPurpose = { ...getMockPurpose(), versions: [mockPurposeVersion] };
+
+  const apiResponse = purposeApi.PurposeVersion.parse(
+    purposeVersionToApiPurposeVersion(mockPurposeVersion)
+  );
+
+  beforeEach(() => {
```

Comment:
@sandrotaje as we discussed, I used `beforeEach` here so that this mock is the default for all tests in the file unless they define new mocks

It doesn't change much, but an advantage is that the final test for 400 for invalid input uses this mock, and not the last defined rejection mock (which might be an exception that is mapped to 400, in which case the invalid input test might succeed silently even if the input is accepted)

cc @MatteoGrilli-quinck